source/common/stats/symbol_table_impl.h

```diff
@@ -400,6 +407,29 @@ class StatNameManagedStorage : public StatNameStorage {
   SymbolTable& symbol_table_;
 };
 
+class StatNameManagedContainer {
+public:
+  explicit StatNameManagedContainer(SymbolTable& symbol_table) : symbol_table_(symbol_table) {}
+  ~StatNameManagedContainer();
+
+  /**
+   * @param name the name to add the container.
+   * @return the StatName held in the container for this name.
+   */
+  StatName add(absl::string_view name);
+
+  /**
+   * @param name the name to add the container.
+   * @return a pointer to the bytes held in the container for this name, suitable for
+   *         using to construct a StatName.
+   */
+  uint8_t* addReturningStorage(absl::string_view name);
```

Comment:
Could you add a note on what is meant by 'returning' here?

Reply:
done.